R/checkers.R

```diff
@@ -45,3 +62,264 @@ check_elements_non_neg <- function(x, arg = "x", call = rlang::caller_env()) {
   }
   invisible()
 }
+
+#' Assert that there is no missignness in a particular vector
+#'
+#' @param x the vector to check
+#' @param arg the name of the vector to check
+#' @param call Calling environment to be passed to [cli::cli_abort()] for
+#' traceback.
+#'
+#' @return NULL, invisibly
+assert_non_missingness <- function(x, arg = "x", call = rlang::caller_env()) {
+  if (checkmate::anyMissing(x)) {
+    cli::cli_abort(
+      c("{.arg {arg}} has missing values",
+        "!" = "All elements of{.arg {arg}} should be present",
+        "i" = "Element(s) {.val {which(is.na(x))}} are missing"
+      ),
+      class = "wwinference_input_data_error",
+      call = call
+    )
+  }
+  invisible()
+}
+
+#' Check that there are no repeated elements in the vector
+#'
+#' @description
+#' This function  checks that the elements of a vector are
+#' not repeated.
+#'
+#' @param x the vector to check
+#' @param arg the name of the vector to check
+#' @param call Calling environment to be passed to [cli::cli_abort()] for
+#' traceback.
+#'
+#' @return NULL, invisibly
+assert_no_repeated_elements <- function(x, arg = "x",
+                                        call = rlang::caller_env()) {
+  duplicates <- duplicated(x)
+  if (any(duplicates)) {
+    cli::cli_abort(
+      c("{.arg {arg}} has more than one element",
+        "i" = c(
+          "Multiple {.arg {arg}} are not currently supported.",
+          "Check that data is from a single location, and if so, provide",
+          "a single count data stream for the population in that location"
+        ),
+        "!" = "Duplicate element(s) index: {.val {which(duplicates)}}"
+      ),
+      call = call,
+      class = "wwinference_input_data_error"
+    )
+  }
+  invisible()
+}
+
+
+
+#' Assert that the arguments are either count type or characters
+#' @description
+#' This is for unique identifiers of groupings, which we will allow to
+#' either be character or integers.
+#'
+#'
+#' @param x Object with type checking
+#' @param arg Name of the argument supplying the object
+#' @param call Calling environment to be passed to [cli::cli_abort()]
+#'
+#' @return NULL, invisibly
+assert_int_or_char <- function(x, arg = "x", call = rlang::caller_env()) {
+  count_or_char_check_result <- checkmate::check_integerish(x) ||
+    checkmate::check_character(x)
```

Comment:
this won't work on failure I think as you can't `||` two strings

Reply:
ooh good check. Hopefully would've caught this in tests. Will come up with a workaround